src/libultra/io/devmgr.c

```diff
@@ -1,97 +1,100 @@
 #include "global.h"
 #include "ultra64/internal.h"
+#include "ultra64/leodrive.h"
 
 void __osDevMgrMain(void* arg) {
-    OSIoMesg* ioMesg;
-    OSMesg sp70;
-    OSMesg sp6C;
-    OSMgrArgs* arg0 = (OSMgrArgs*)arg;
-    __OSTranxInfo* transfer;
-    __OSBlockInfo* block;
-    s32 phi_s2;
-    s32 phi_s0;
-    u32 sp54;
-
-    ioMesg = NULL;
+    OSIoMesg* ioMesg = NULL;
+    OSMesg em;
+    OSMesg dummy;
+    s32 ret;
+    OSDevMgr* dm = (OSDevMgr*)arg;
+    s32 messageSend;
 
     while (true) {
-        osRecvMesg(arg0->cmdQueue, (OSMesg*)&ioMesg, OS_MESG_BLOCK);
+        osRecvMesg(dm->cmdQueue, (OSMesg*)&ioMesg, OS_MESG_BLOCK);
         if ((ioMesg->piHandle != NULL) && (ioMesg->piHandle->type == DEVICE_TYPE_64DD) &&
             ((ioMesg->piHandle->transferInfo.cmdType == 0) || (ioMesg->piHandle->transferInfo.cmdType == 1))) {
-            transfer = &ioMesg->piHandle->transferInfo;
-            block = &transfer->block[transfer->blockNum];
-            transfer->sectorNum = -1;
-            if (transfer->transferMode != 3) {
-                block->dramAddr = (void*)((u32)block->dramAddr - block->sectorSize);
+            __OSBlockInfo* blockInfo;
+            __OSTranxInfo* info;
+
+            info = &ioMesg->piHandle->transferInfo;
+            blockInfo = &info->block[info->blockNum];
+
+            info->sectorNum = -1;
+            if (info->transferMode != 3) {
+                blockInfo->dramAddr = (void*)((u32)blockInfo->dramAddr - blockInfo->sectorSize);
             }
 
-            phi_s2 = ((transfer->transferMode == 2) && (ioMesg->piHandle->transferInfo.cmdType == 0)) ? 1 : 0;
+            messageSend = ((info->transferMode == 2) && (ioMesg->piHandle->transferInfo.cmdType == 0)) ? true : false;
```

Comment:
honestly `messageSend` as a boolean is more confusing to me than a 0/1 or is there something I'm missing

Reply:
I can see that it can be a bit confusing since it was written weirdly, (`messageSend != true`) but I don't think that disqualifies it from being a boolean?

Reply:
well usually we still boolify `== true`-using stuff when it's otherwise mostly used as a bool, here the only use is `== true`, but whatever

My comment is more that, what does "messageSend == true" mean? the message will be / has been sent ? idk it just looks like a "exit loop" flag to me and for some reason using true/false makes it harder to read

but if it's just me whatever it's not like this is important